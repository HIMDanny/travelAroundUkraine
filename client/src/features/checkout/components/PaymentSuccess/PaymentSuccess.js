/* eslint-disable react/prop-types */
import React,{memo} from 'react';
import { Typography, Box } from '@mui/material';
import CheckCircleOutlineRoundedIcon from '@mui/icons-material/CheckCircleOutlineRounded';
import { useSelector, useDispatch } from 'react-redux';
import { setIsModalOpen, increaseStep } from '../../../../store/slices/orderSlice/orderSlice';
import { AlertModal } from '../../../../components';

const PaymentSuccess = () => {
  const { isModalOpen, orderInfo } = useSelector((store) => store.order);
  const dispatch = useDispatch();

  const handleCloseBttn = () => {
    dispatch(setIsModalOpen(false));
    dispatch(increaseStep());
  };

  const titleJsx = (
    <Box display="flex" justifyContent="center" alignItems="center" columnGap="5px">
      <Typography variant="h2" sx={{ color: 'success.main', mb: 0 }}>
        Payment Successfull
      </Typography>
      <CheckCircleOutlineRoundedIcon fontSize="large" color="success" />
    </Box>
  );

  return (
    <AlertModal
      open={isModalOpen}
      onClose={handleCloseBttn}
      onSubmit={handleCloseBttn}
      title={titleJsx}
      disableCancelButton
      submitButtonText="Great !"
      success
    >
      Your order number #{orderInfo?.orderNo} will be shipped shortly.
    </AlertModal>
  );
};

export default memo(PaymentSuccess);
