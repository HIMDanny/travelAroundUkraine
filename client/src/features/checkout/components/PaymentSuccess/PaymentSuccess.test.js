/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import PaymentSuccess from './PaymentSuccess';

jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/Grid', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Dialog', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogContent', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogActions', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogTitle', () => ({ children }) => <h3>{children}</h3>);
jest.mock('@mui/icons-material/CheckCircleOutlineRounded', () => () => <div>ICON</div>);

const steps = ['User Details', 'Shipping Address', 'Payment Details'];

describe('PaymentSuccess snapshot test', () => {
  test('should PaymentSuccess render null', () => {
    const { asFragment } = render(<PaymentSuccess steps={steps} activeStep={0} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should PaymentSuccess render modal window', () => {
    const { asFragment } = render(<PaymentSuccess steps={steps} activeStep={3} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
