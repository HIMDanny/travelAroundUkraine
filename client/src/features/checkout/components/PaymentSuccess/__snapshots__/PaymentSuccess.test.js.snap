// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSuccess snapshot test should PaymentSuccess render modal window 1`] = `
<DocumentFragment>
  <div>
    <h3>
      <div>
        <p>
          Payment Successfull
        </p>
        <div>
          ICON
        </div>
      </div>
    </h3>
    <div>
      <p />
      <p>
        Your order number #425435 will be shipped shortly.
      </p>
      <p />
    </div>
    <div>
      <button>
        Great !
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PaymentSuccess snapshot test should PaymentSuccess render null 1`] = `<DocumentFragment />`;
